packages/shylu/shylu_dd/frosch/src/SchwarzOperators/FROSch_HarmonicCoarseOperator_def.hpp

```diff
@@ -762,142 +782,226 @@ namespace FROSch {
         }
         return mVPhi;
     }
-    // REPMAP-------------------------------------
-    template<class SC,class LO, class GO, class NO>
-    int HarmonicCoarseOperator<SC,LO,GO,NO>::buildGlobalGraph(Teuchos::RCP<DDInterface<SC,LO,GO,NO> > theDDInterface)
+
+    // Should we move this to Tools?
+    template <class SC,class LO,class GO,class NO>
+    typename HarmonicCoarseOperator<SC,LO,GO,NO>::ConstXMatrixPtr HarmonicCoarseOperator<SC,LO,GO,NO>::removeCouplingBetweenDofs(ConstXMatrixPtr matrix,
+                                                                                                                                 ConstXMapPtr map,
+                                                                                                                                 TwoDArray<int> &couplingIDsToRemove)
```

Comment:
should the type of `TwoDArray` be `LO`?

Reply:
The indices are small number indicating the the number of the block and the dof ID which should be removed. This should be more efficient than specifying each LO or GO which should be removed. Therefore, `int` should be fine.